tracer/src/Datadog.Trace/Sampling/ISpanSampler.cs

```diff
@@ -0,0 +1,26 @@
+// <copyright file="ISpanSampler.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+namespace Datadog.Trace.Sampling
+{
+    /// <summary>
+    ///     Represents the functionality to sample individual spans instead of entire traces.
+    /// </summary>
+    internal interface ISpanSampler
+    {
+        /// <summary>
+        ///     Makes a <see cref="SamplingDecision"/> for the given <paramref name="span"/>.
+        /// </summary>
+        /// <param name="span">The <see cref="Span"/> to sample.</param>
+        void MakeSamplingDecision(Span span);
```

Comment:
From the names of these methods, I would expect `MakeSamplingDecision()` to return the result (`bool`?) and not modify the span. Then the caller would do something like:
```csharp
if(sampler.MakeSamplingDecision(span))
{
    sampler.SetTags(span);
}
```

(Yes, `TraceSampler` also add tags, and I wish we would have done that differently. I may have to refactor that soon, anyway.)

Separating these two actions (making the sampling decision vs tagging the span) also lets you unit test them separately. Think something like that doesn't involve tags:

```csharp
sampler.MakeSamplingDecision(keptSpan).Should().BeTrue();

sampler.MakeSamplingDecision(droppedSpan).Should().BeFalse();
```

Reply:
Just a thought. Obvious needs more work since you also need to pass the `ISpanSamplingRule` around somehow 😅